File: ProduccionUnificado/backend/dump_mappings.cs
using System;
using System.Linq;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore;
using TiempoProcesos.API.Data;
using TiempoProcesos.API.Models;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Configuration;

namespace MappingDumper
{
    class Program
    {
        static void Main(string[] args)
        {
            var config = new ConfigurationBuilder()
                .SetBasePath(System.IO.Directory.GetCurrentDirectory())
                .AddJsonFile("appsettings.json")
                .Build();

            var optionsBuilder = new DbContextOptionsBuilder<AppDbContext>();
            optionsBuilder.UseNpgsql(config.GetConnectionString("DefaultConnection"));

            using (var context = new AppDbContext(optionsBuilder.Options))
            {
                var maquinas = context.Maquinas.Select(m => new { m.Id, m.Nombre }).ToList();
                var usuarios = context.Usuarios.Select(u => new { u.Id, u.Nombre }).ToList();
                var codigos = context.CodigosDesperdicio.Select(c => new { c.Id, c.Codigo, c.Descripcion }).ToList();

                System.IO.File.WriteAllText("mappings.json", System.Text.Json.JsonSerializer.Serialize(new { maquinas, usuarios, codigos }, new System.Text.Json.JsonSerializerOptions { WriteIndented = true }));
                Console.WriteLine("Mappings dumped to mappings.json");
            }
        }
    }
}

File: ProduccionUnificado/backend/dump_mappings.js
const { Client } = require('pg');
const fs = require('fs');

async function main() {
    const client = new Client({
        host: 'localhost',
        port: 5432,
        database: 'TiemposProcesos',
        user: 'postgres',
        password: '@L3ph2026',
    });

    try {
        await client.connect();
        
        const maquinas = await client.query('SELECT "Id", "Nombre" FROM "Maquinas"');
        const usuarios = await client.query('SELECT "Id", "Nombre" FROM "Usuarios"');
        const codigos = await client.query('SELECT "Id", "Codigo", "Descripcion" FROM "CodigosDesperdicio"');

        const mappings = {
            maquinas: maquinas.rows,
            usuarios: usuarios.rows,
            codigos: codigos.rows
        };

        fs.writeFileSync('mappings.json', JSON.stringify(mappings, null, 2));
        console.log('Mappings dumped to mappings.json');
    } catch (err) {
        console.error('Error connecting to database:', err);
    } finally {
        await client.end();
    }
}

main();
